scripts/mk_all.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Makes a file all.lean in all subfolders of src/ importing all files in that folder, including subfolders.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
What is the effect of this line? I can't parse it.

Reply:
It `cd`s to the directory in which the called script is located (it finds this directory using `dirname`) and then calls `pwd`. I don't know what the point of the `cd + pwd` instead of just directly using the result from `dirname` is.

Reply:
:bulb:

Reply:
This line of code is supposed to get the absolute path of the script directory (I found it online by Googling).
Just doing `$( dirname "${BASH_SOURCE[0]}" )` will often give you a relative path to the script directory, which I was a problem (at least when I was working on the script, maybe not with the final version)

Reply:
I just checked, the relative path causes issues in `mk_all.sh` because of my `cd` in the for loop. In `rm_all.sh` I could simplify it, if you think that is desirable.